src/analyses/baseInvariant.ml

```diff
@@ -64,7 +64,7 @@ struct
 
   let refine_lv_fallback ctx a gs st lval value tv =
     if M.tracing then M.tracec "invariant" "Restricting %a with %a\n" d_lval lval VD.pretty value;
-    let addr = eval_lv a gs st lval in
+    let addr = eval_lv ~ctx lval in
```

Comment:
This is also semantically different now because maybe the `st` argument it previously used is something that keeps changing during refinement?

Other places below beg the same question.